Changed package to callback abstract

XedinUnknown opened the pull request:
Aims to incorporate solution for #2.

XedinUnknown committed:
Merged upstream

mecha reviewed (changes requested):
Kindly remove `.idea/misc.xml` from version control and rename `.idea/callback-aware-trait.iml` to `.idea/callback-abstract.iml`.

Also, I've been noticing that `.idea/php.xml` does not use the `$PROJECT_DIR$` placeholder. Maybe we ought to stop committing that and adding it to `.gitignore`.

The only thing that PHPStorm needs to recognize that a folder is a project is the `.idea/*.iml` file.

XedinUnknown committed:
Removed `misc.xml`

XedinUnknown committed:
Renamed PHPS project settings file

mecha commented:
@XedinUnknown Actually, my projects' `.idea/php.xml` files are correctly using the `$PROJECT_DIR$` placeholder. Perhaps your paths are incorrect?

XedinUnknown committed:
Added dev dep on stringable interface

XedinUnknown committed:
Added `InvokeCallableCapableTrait`

This completes the move described in dhii/code-map-command-invoker-abstract#2.

XedinUnknown commented:
@mecha, not sure exactly. Just removed the `misc.xml` for now.

XedinUnknown committed:
`_invokeCallable()` now allowed to throw root `Exception`

XedinUnknown committed:
Added `ArgsAwareTrait` and test

This also addresses dhii/args-aware-trait#2.

XedinUnknown committed:
`_invokeCallable()` now throws `InvocationFailureException` for all errors during invocation

Therefore, it is now possible to discriminate between callable logic
errors, and errors raised by the callable itself.

XedinUnknown committed:
Exceptions in callables must now be wrapped in a standard exception

XedinUnknown committed:
Exceptions raised by callables now wrapped in standard exception

XedinUnknown committed:
Code standards fixed

XedinUnknown committed:
Now depending on invocable interface for testing

XedinUnknown committed:
Code standards fixed

XedinUnknown committed:
Added test for `InvokeCallbackCapableTrait`

XedinUnknown committed:
Added `InvokeCallbackCapableTrait`

XedinUnknown committed:
Dependencies updated

XedinUnknown committed:
Fixed paths in project config

XedinUnknown committed:
Merge branch 'develop' into task/2_change-to-callback-abstract

Conflicts:
	.idea/modules.xml
	.idea/php.xml
	composer.lock
	test/functional/CallbackAwareTraitTest.php

mecha commented on src/InvokeCallbackCapableTrait.php:
```diff
@@ -0,0 +1,132 @@
+             * `_invokeCallback()` would throw `InvalidArgumentException`
+             * is if the callback is wrong. But we cannot let it bubble
+             * up, because it is not an argument to this method. Therefore,
+             * catch it, wrap, and trow a more appropriate exception.
```
Typo in _"trow"_

mecha commented on src/InvokeCallbackCapableTrait.php:
```diff
@@ -0,0 +1,132 @@
+     *
+     * @since [*next-version*]
+     *
+     * @return callable
```
Missing return description.

mecha reviewed (changes requested):
Over all, I love this. I'm only requesting a couple of small things relating to docs.

XedinUnknown committed:
Fixed typo in inline doc

XedinUnknown committed:
Added missing return doc
